src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/HttpGoogleServicesDAO.scala

```diff
@@ -324,6 +324,31 @@ class HttpGoogleServicesDAO(
     }
   }
 
+  override def getBucketUsage(bucketName: String, maxResults: Option[Long]): Future[BigInt] = {
+    def sumItemSizes(items: java.util.List[StorageObject]) = {
+      (BigInt(0) /: items) { (sum, o) => sum + o.getSize }
+    }
+
+    def processPage(pageToken: Option[String] = None): Future[BigInt] = {
+      val fetcher = getStorage(getBucketServiceAccountCredential).objects().list(bucketName)
+      if (maxResults.isDefined) fetcher setMaxResults maxResults.get
+      if (pageToken.isDefined) fetcher setPageToken pageToken.get
+
+      retryWithRecoverWhen500orGoogleError(() => {
+        val result = executeGoogleRequest(fetcher)
+        result match {
+          case r if r.getItems == null => BigInt(0)
+          case r if r.getNextPageToken == null => sumItemSizes(r.getItems)
+          case r => sumItemSizes(r.getItems) + Await.result(processPage(Some(result.getNextPageToken)), Duration.Inf)
```

Comment:
Await.result is a blocking task so generally we only use it in tests and in production code that is very rarely used by admins. this endpoint could be called every time a user loads a workspace in the UI so we want to avoid that here